src/BotwinModule.cs

```diff
@@ -1,131 +1,119 @@
-using System.Threading;
-
 namespace Botwin
 {
     using System;
     using System.Collections.Generic;
-    using System.Dynamic;
-    using System.IO;
-    using System.Linq;
     using System.Threading.Tasks;
     using Microsoft.AspNetCore.Http;
     using Microsoft.AspNetCore.Routing;
 
     public class BotwinModule
     {
-        private readonly string basePath;
+        public readonly List<(string verb, string path, RequestDelegate handler)> Routes;
 
-        public List<(string verb, string path, RequestDelegate handler)> Routes =
-            new List<(string verb, string path, RequestDelegate handler)>();
+        private readonly string basePath;
 
-        public Func<HttpContext, Task<bool>> Before { get; set; }
+        public Func<HttpContext, Task<bool>> Before { get; protected set; }
 
-        public RequestDelegate After { get; set; }
+        public RequestDelegate After { get; protected set; }
 
         protected BotwinModule() : this(string.Empty)
         {
         }
 
         protected BotwinModule(string basePath)
         {
+            this.Routes = new List<(string verb, string path, RequestDelegate handler)>();
             var cleanPath = this.RemoveStartingSlash(basePath);
             this.basePath = this.RemoveEndingSlash(cleanPath);
         }
 
-        public void Get(string path, Func<HttpRequest, HttpResponse, RouteData, Task> handler)
+        protected void Get(string path, Func<HttpRequest, HttpResponse, RouteData, Task> handler)
         {
-            RequestDelegate requestDelegate = httpContext =>
-                handler(httpContext.Request, httpContext.Response, httpContext.GetRouteData());
-            this.Get(path, requestDelegate);
+            Task RequestDelegate(HttpContext httpContext) => handler(httpContext.Request, httpContext.Response, httpContext.GetRouteData());
```

Comment:
What is this?

Reply:
Oops, screwed up the rebase.

Reply:
An inline function. C# 7 thing. Makes working with functions and the like way simpler

Reply:
Adds no value here IMO

Reply:
There was a reason I did this, if you asked me 2 days ago I would have told you

Reply:
Perf: https://github.com/dotnet/roslyn/issues/3911